### DISCUSSION TITLE
how to understand the transmission time on air?

### DISCUSSION THREAD

**Discussion by hansT-T** [Lora-net/LoRaMac-node#1547] [Q&A]
hey guys，
i am studying LoRa recently.
In sx127x Datasheet, it says the lora phy frame is consist of preamble, header, payload, and payload crc.
especially, the header is within a code rate of 4/8.
But in the calculation of Time on Air, 
it says, nPayload = 8 + max(ceil((8PL-4SF+28+16CRC-20IH)/4(SF-2DE))*(CR+4),0)
how to understand the calculation of the payload?
is the 20IH means the length of the header? 
but header is within a code rate of 4/8, isn't it?
why we calculate the header and payload together within the same code rate?
I would be grateful if someone could help me.
![0R(7OSHG)3KB_S QM$ ~12](https://github.com/Lora-net/LoRaMac-node/assets/118200147/6550f6b9-b542-4bcc-accb-3665c536c81a)
![AA6F3 7CFI8)KQYLLM}MT{A](https://github.com/Lora-net/LoRaMac-node/assets/118200147/0e1aefd3-95d7-4990-b542-de58c6adea0a)

**Comment by mluis1** [comment]
Thanks for the question, however your question does not look to be an issue or question specifically related to this project.
Your question is a general question about the LoRa modulation and radio chips datasheets.

I would recommend that you ask your question on [Semtech LoRa Developers Forum](https://forum.lora-developers.semtech.com/) being a more suitable place to ask questions.
Please use the forum search as I am pretty sure that your or similar question has already been answered.
